src/robusta/core/discovery/top_service_resolver.py

```diff
@@ -51,20 +53,52 @@ def store_cached_resources(cls, resources: List[TopLevelResource]):
     # TODO remove this guess function
     # temporary try to guess who the owner service is.
     @classmethod
-    def guess_service_key(cls, name: str, namespace: str) -> str:
-        resource = cls.guess_cached_resource(name, namespace)
+    def guess_service_key(cls, name: str, namespace: str, kind: str) -> str:
+        resource = cls.guess_cached_resource(name, namespace, kind=kind)
         return resource.get_resource_key() if resource else ""
 
+    @classmethod
+    def get_pod_owner_reference(cls, name: str, namespace: str) -> OwnerReference:
+        robusta_pod = RobustaPod.find_pod(name, namespace)
```

Comment:
@Avi-Robusta @ganeshrvel I think doing this is a very bad idea.
This will make every call on a pod, to call the API server.
This is called for every finding.
It must be done within memory, without calling the k8s api server

Reply:
Also arik mentioned even if we get the pod that we only have the replicaset owner,  since we want the owner of the replicaset like the deployment or job